examples/seismic/tutorials/06_skimage_tv.ipynb

```diff
@@ -0,0 +1,416 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "collapsed": true
+   },
+   "source": [
+    "# 06 - FWI with total variation (TV) minimization as constraints"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Introduction\n",
+    "\n",
+    "There is a lot of research on regularization to improve the quality of the final result beyond the simple box constraints implemented in the previous tutorials. In this tutorial we look at how more advanced FWI techniques such as [total variation denoising]( https://en.wikipedia.org/wiki/Total_variation_denoising) applied as a constraint can be implemented using [Devito](http://www.opesci.org/devito-public) and [scipy.optimize.minimize](https://docs.scipy.org/doc/scipy/reference/generated/scipy.optimize.minimize.html) using [scikit-image](http://scikit-image.org/). This is a variant of the FWI with TV constrains algorithm described in [Peters and Herrmann 2017](https://doi.org/10.1190/tle36010094.1).\n",
+    "\n",
+    "[Dask](https://dask.pydata.org/en/latest/#dask) is also used here to speed up the examples.\n",
+    "\n",
+    "This tutorial uses the same synthetic datasets and model setup as the previous two tutorals, so check back if you get lost on parts of the code specific to Devito, SciPy.optimize or Dask. "
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Setting up (synthetic) data\n",
+    "We are going to set up the same synthetic test case as for the previous tutorial (refer back for details)."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "metadata": {
+    "collapsed": true
+   },
+   "outputs": [],
+   "source": [
+    "#NBVAL_IGNORE_OUTPUT\n",
+    "from examples.seismic import Model, demo_model\n",
+    "\n",
+    "# Define the grid parameters\n",
+    "def get_grid():\n",
+    "    shape = (101, 101)    # Number of grid point (nx, nz)\n",
+    "    spacing = (10., 10.)  # Grid spacing in m. The domain size is now 1km by 1km\n",
+    "    origin = (0., 0.)     # Need origin to define relative source and receiver locations\n",
+    "\n",
+    "    return shape, spacing, origin\n",
+    "\n",
+    "# Define the test phantom; in this case we are using a simple circle\n",
```

Comment:
Same as previous tutorial. Also, is there a way you can import the common pieces of code from the previous tutorial? i.e. in order to avoid duplicate code

Reply:
I hadn't noticed this. Please, yes

Reply:
Agreed - but that will be a separate PR.

Reply:
All things that go in separate PRs should have a corresponding GitHub issue